test/fshtypes/Instance.test.ts

```diff
@@ -15,6 +15,26 @@ describe('Instance', () => {
     });
   });
 
+  describe('#id', () => {
+    it('should return an id set by a string assignment rule', () => {
+      const p = new Instance('MyInstance');
+      const idRule = new AssignmentRule('id');
+      idRule.value = 'different-id';
+      p.rules.push(idRule);
+      expect(p.id).toBe('different-id');
+    });
+
+    it('should not return an id set by an instance assignment rule', () => {
+      const p = new Instance('MyInstance');
+      const idRule = new AssignmentRule('id');
+      idRule.value = 'different-id';
+      idRule.isInstance = true;
+      p.rules.push(idRule);
+      // the id is still the default
+      expect(p.id).toBe('MyInstance');
+    });
```

Comment:
Cool. Can we assert no errors in the first test and the expected error in the second test?

Reply:
There aren't any expected errors in either test, since nothing is being exported. While it's unusual to assign an Instance here, it's allowed. Although, I just tested out some FSH that does this, and the output is not quite right:
```
Instance: MyObs
InstanceOf: Observation
Usage: #example
* status = #final
* code = #123
* id = SomeId

Instance: SomeId
InstanceOf: http://hl7.org/fhir/StructureDefinition/id
Usage: #inline
* value = "my-obs-id"
```

The Instance gets assigned using its object structure, the resulting file for `MyObs` is named `Observation-[object Object].json`, and SUSHI emits an error: `The string "[object Object]" does not represent a valid FHIR id.` We might need to investigate assignment of inline primitive instances.

Reply:
Yeah, primitive instances in general are a pretty weird case. I'm not very surprised that we don't handle them well in some (or all?) cases.